### PR TITLE
ollama: add support for moderation model

### PR DESCRIPTION
**Pull request by rockwotj**
- **ollama: support saving the prompt as metadata**
- **ollama: add ollama_moderation for llama-guard and shieldgemma**

This allows us to classify LLM responses for safety.

### CODE REVIEW

File: internal/impl/ollama/moderation_processor.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		return nil, err
+	}
+	r, err := o.response.TryString(msg)
+	if err != nil {
+		return nil, err
+	}
+	g, err := o.generateCompletion(ctx, p, r)
+	if err != nil {
+		return nil, err
+	}
```
Comment (mihaitodor) [inline internal/impl/ollama/moderation_processor.go:123]: Might be worth adding some extra context to these errors.

**Review by mihaitodor — APPROVED** [review]
Nice job @rockwotj! Just left a few nits. Feel free to :shipit:

File: internal/impl/ollama/moderation_processor.go (hunk 1 above)
Comment (rockwotj) [reply internal/impl/ollama/moderation_processor.go]: Done!

File: internal/impl/ollama/chat_processor.go [hunk 2]
```diff
@@ -121,6 +122,9 @@ For more information, see the https://github.com/ollama/ollama/tree/main/docs[Ol
 				Optional().
 				Advanced().
 				Description(`Sets the stop sequences to use. When this pattern is encountered the LLM stops generating text and returns the final response.`),
+			service.NewBoolField(ocpFieldEmitPromptMetadata).
+				Default(false).
+				Description(`If enabled the prompt is saved as @prompt metadata on the output message.`),
```
[the remark this answers is not in the document]
Comment (rockwotj) [reply internal/impl/ollama/chat_processor.go]: Thanks - yes

File: internal/plugins/info.csv [hunk 3]
```diff
@@ -155,6 +155,7 @@ ockam_kafka               ,input     ,ockam_kafka               ,0.0.0   ,commun
 ockam_kafka               ,output    ,ockam_kafka               ,0.0.0   ,community  ,n          ,n     ,n
 ollama_chat               ,processor ,ollama_chat               ,4.32.0  ,enterprise ,n          ,n     ,y
 ollama_embeddings         ,processor ,ollama_embeddings         ,4.32.0  ,enterprise ,n          ,n     ,y
+ollama_moderation         ,processor ,ollama_moderation         ,0.0.0   ,enterprise ,n          ,n     ,y
```
[the remark this answers is not in the document]
Comment (rockwotj) [reply internal/plugins/info.csv]: Done

**Pull request merged**